quack_app/test/screens/notification/notification_screen_test.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:golden_toolkit/golden_toolkit.dart';
+import 'package:quack_app/constants/constants.dart';
+import 'package:quack_app/core/notification_data.dart';
+import 'package:quack_app/main.dart';
+import 'package:quack_app/screens/notification/notification_screen.dart';
+
+void main() {
+  group("Notification Screen Test", () {
+    NotificationData fakeNotificationData = NotificationData();
+    fakeNotificationData.getExistingNotifications();
+
+    MyApp app = const MyApp();
+    testGoldens("basic_view", (WidgetTester tester) async {
+      await loadAppFonts();
+      await tester.pumpWidget(app);
+      await tester.tap(find.byIcon(Constants.kNavBarNotificationsIcon));
+      await tester.pumpAndSettle(const Duration(seconds: 1));
+
+      await expectLater(
+          find.byType(NotificationScreen),
+          matchesGoldenFile('goldens/basic_view.png'));
+    });
+
+    testGoldens("delete_notification", (WidgetTester tester) async {
+      await tester.pumpWidget(app);
+      await tester.tap(find.byIcon(Constants.kNavBarNotificationsIcon));
+      await tester.pumpAndSettle(const Duration(seconds: 1));
+
+      await tester.tap(find.text("Delete").first);
+      await loadAppFonts();
+      await tester.pumpAndSettle(const Duration(seconds: 1));
+      await expectLater(
+          find.byType(NotificationScreen), matchesGoldenFile('goldens/delete_notification_1.png'));
+      int remaining = fakeNotificationData.currentNotifications.length;
```

Comment:
You may want to take the same approach here. Find the length of total notifications first before deletion, then expect `length - 2` instead of hard coding it here.